### PR TITLE
Initial implementation of the connector.

### PR DESCRIPTION
**Pull request by hemikak**
The following PR contains the required synapse mediators to make a call to an AS400 server program.

### CODE REVIEW

File: src/main/java/org/wso2/carbon/connector/pcml/AS400Constants.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+/**
+ * Constants for the connector.
+ */
+public class AS400Constants {
+    // Properties stored in the message context.
+    public static final String AS400_INSTANCE = "as400Instance";
+    public static final String AS400_SOCKET_PROPERTIES = "as400SocketProperties";
+    public static final String AS400_CONNECTION_POOL_MAP = "as400ConnectionPoolMap";
+    // Parameters of the initializing mediator.
+    public static final String AS400_INIT_SYSTEM_NAME = "systemName";
```
Comment (malakasilva) [inline src/main/java/org/wso2/carbon/connector/pcml/AS400Constants.java]: Can we prefix something specific to connector for messagecontext properties.

File: src/main/java/org/wso2/carbon/connector/pcml/AS400CallProgram.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        SynapseLog log = getLog(messageContext);
+        AS400 as400 = null;
+        try {
+            Object as400InstanceProperty = messageContext.getProperty(AS400Constants.AS400_INSTANCE);
+            if (null != as400InstanceProperty) {
+                as400 = (AS400) as400InstanceProperty;
+            } else {
+                throw new AS400PCMLConnectorException("Unable to find an AS400 instance to call program. Use the " +
```
Comment (hemikak) [inline src/main/java/org/wso2/carbon/connector/pcml/AS400CallProgram.java]: Fixed in https://github.com/wso2-extensions/esb-connector-pcml/pull/3/files#diff-662ab4a619a4739c3828ff890f9fe024R65

File: src/main/java/org/wso2/carbon/connector/pcml/AS400Constants.java (hunk 1 above)
Comment (hemikak) [inline src/main/java/org/wso2/carbon/connector/pcml/AS400Constants.java]: Fixed in https://github.com/wso2-extensions/esb-connector-pcml/pull/3/files#diff-aa3a15281a8a00341109fa1ef26ab435R30

**Comment by hemikak** [comment]
Refer to https://github.com/wso2-extensions/esb-connector-pcml/pull/3 for the latest code.

**Pull request closed without merging**